drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c

```diff
@@ -1020,13 +1020,27 @@ static int vidioc_try_fmt_vid_cap(struct file *file, void *priv,
 			      1, 0);
 	f->fmt.pix.bytesperline = f->fmt.pix.width * mfmt->ybbp;
 	if (!mfmt->remove_padding) {
-		int align_mask = ((32 * mfmt->depth) >> 3) - 1;
-		/* GPU isn't removing padding, so stride is aligned to 32 */
-		f->fmt.pix.bytesperline =
-			(f->fmt.pix.bytesperline + align_mask) & ~align_mask;
+		if (mfmt->depth == 24) {
+			/*
+			 * 24bpp is a pain as we can't use simple masking.
+			 * Min stride is width aligned to 16, times 24bpp.
+			 */
+			f->fmt.pix.bytesperline =
+				((f->fmt.pix.width + 15) & ~15) * mfmt->ybbp;
```

Comment:
Is there any reason not to replace mfmt->ybbp to 3? I find it clearer when read with the comment.

Reply:
James had made the same comment verbally.
mfmt->ybpp in this case will always be 3, therefore it can be replaced.